### PR TITLE
Support varbinary as bytea for PostgreSQL

### PR DESCRIPTION
**Pull request by findepi**
Based on @electrum's https://github.com/prestodb/presto/pull/9413

### CODE REVIEW

File: presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java [hunk 1]
```diff
@@ -81,6 +87,28 @@ public void testBasicTypes()
                 .execute(getQueryRunner(), prestoCreateAsSelect("test_basic_types"));
     }
 
+    @Test
+    public void testVarbinary()
+    {
+        DataTypeTest.create()
```
Comment (electrum) [inline presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java]: The test cases are redundant. We could move them to a method:
```java
DataTypeTest varbinaryTest = DataTypeTest.create();
addVarbinaryTests(varbinaryDataType(), "hello".getBytes(UTF_8));
varbinaryTest.execute(getQueryRunner(), prestoCreateAsSelect("test_varbinary"));

File: presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java [hunk 2]
```diff
@@ -201,6 +229,15 @@ private void testUnsupportedDataType(String databaseDataType)
         }
     }
 
+    private static DataType<byte[]> byteaDataType()
+    {
+        return DataType.dataType(
+                "bytea",
+                VARBINARY,
+                bytes -> format("bytea E'\\\\x%s'", base16().encode(requireNonNull(bytes, "bytes is null"))),
```
Comment (electrum) [inline presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java]: No need for `requireNonNull` here as the `encode` method will throw NPE.

The rationale for null checks in constructors and other places is to ensure the object invariants and to have the exception occur at the right time, when you know who provided the null. In this case, the caller will get an NPE with the proper stack trace.

Occasionally, we might want the explicit check to provide a better message, but that shouldn't be needed here as there's only one object.

Now, if `bytes` was an argument to this method and was being put directly into the lambda capture, we'd want a null check to ensure the capture is valid.

File: presto-tests/src/main/java/com/facebook/presto/tests/datatype/DataType.java [hunk 3]
```diff
@@ -135,6 +143,14 @@ private static String quote(String value)
         return "'" + value + "'";
     }
 
+    /**
+     * Formats bytes using SQL standard format for binary string literal
+     */
+    public static String binaryLiteral(byte[] value)
+    {
+        return "X'" + base16().encode(requireNonNull(value, "value is null")) + "'";
```
Comment (electrum) [inline presto-tests/src/main/java/com/facebook/presto/tests/datatype/DataType.java]: Same as above

File: presto-tests/src/main/java/com/facebook/presto/tests/datatype/DataTypeTest.java [hunk 4]
```diff
@@ -51,8 +51,11 @@ public void execute(QueryRunner prestoExecutor, DataSetup dataSetup)
         try (TestTable testTable = dataSetup.setupTestTable(unmodifiableList(inputs))) {
             MaterializedResult materializedRows = prestoExecutor.execute("SELECT * from " + testTable.getName());
             assertEquals(materializedRows.getTypes(), expectedTypes);
-            MaterializedRow row = getOnlyElement(materializedRows);
-            assertEquals(row.getFields(), expectedResults);
+            List<Object> actualResults = getOnlyElement(materializedRows).getFields();
+            assertEquals(actualResults.size(), expectedResults.size(), "lists don't have the same size");
+            for (int i = 0; i < expectedResults.size(); i++) {
+                assertEquals(actualResults.get(i), expectedResults.get(i), format("Element %s", i));
```
Comment (electrum) [inline presto-tests/src/main/java/com/facebook/presto/tests/datatype/DataTypeTest.java]: Nit: I'd make this
```java
assertEquals(..., "Element " + i);

File: presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java (hunk 1 above)
Comment (electrum) [reply presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java]: If we add a new `apply(Consumer<DataTypeTest>)` method to `DataTypeTest`, we can keep the builder shape:
```java
DataTypeTest.create()
        .apply(TestPostgreSqlTypeMapping::addVarbinaryTests)
        .execute(getQueryRunner(), prestoCreateAsSelect("test_varbinary"));

Comment (findepi) [reply presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java]: Done. Extracted test cases to single method.

**Pull request merged**